Framework/Kernel/src/ConfigService.cpp

```diff
@@ -325,6 +325,52 @@ void ConfigServiceImpl::setBaseDirectory() {
 #endif
 }
 
+namespace {
+// look for specific keys and throw an exception if one is found
+std::string checkForBadConfigOptions(const std::string &filename,
+                                     const std::string &propertiesString) {
+  std::stringstream stream(propertiesString);
+  std::stringstream resultPropertiesString;
+  std::string line;
+  int line_num = 0;
+  while (std::getline(stream, line)) {
+    line_num += 1; // increment early
+    bool is_ok = true;
+
+    // Check for common errors. Empty lines are ok, things that are a key
+    // without a value are a critical failure. Forbidden keys are just commented
+    // out.
+    if (line.empty() || (line[0] == '#')) {
+      // do nothing
+    } else if (line.find('=') == std::string::npos) {
```

Comment:
Don't do this. A line that just contain a space or `" # comment"` will not allow to start mantid